olive/passes/onnx/nvmo_quantization.py

```diff
@@ -0,0 +1,116 @@
+        """,
+    ),
+    "batch_size": PassConfigParam(
+        type_=int,
+        default_value=1,
+        description="""
+            Batch size for calibration, only used if dataloader_func is provided.
+        """,
+    ),
+    "calib_size": PassConfigParam(
+        type_=int,
+        default_value=64,
+        description="""
+            Calibration data size, only used if dataloader_func is provided.
+        """,
+    ),
+    "dataloader_func": PassConfigParam(
+        type_=Union[Callable, str],
+        category=ParamCategory.OBJECT,
+        description="""
+            Function/function name to generate dataloader for calibration,
+            required if quant_mode is 'static' and data_config is None.
+        """,
+    ),
+    "dataloader_func_kwargs": PassConfigParam(
+        type_=Dict[str, Any],
+        description="Keyword arguments for dataloader_func.",
+    ),
+}
+
+
+class NVModelOptQuantization(Pass):
+
+    # set this to True if the pass has parameters that are functions or objects and the user script is required
+    # to import the module containing the function or object
+    _requires_user_script: bool = True
+
+    @classmethod
+    def _default_config(cls, accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigParam]:
+        config = {
+            "precision": PassConfigParam(
+                type_=str,
+                default_value="int4",
+                searchable_values=Categorical(["fp8", "int8", "int4"]),
+                description="""
+                    NVModelOpt Quantization mode.
+                """,
+            ),
+            "algorithm": PassConfigParam(
+                type_=str,
+                default_value="RTN",
+                searchable_values=Categorical(["RTN", "AWQ"]),
+                description="""
+                    Algorithm of weight only quantization. Support 'RTN' and 'AWQ'.
+                """,
+            ),
+        }
+
+        config.update(deepcopy(_dataloader_config))
+        return config
+
+    def _run_for_config(
+        self, model: OliveModelHandler, data_root: str, config: Dict[str, Any], output_model_path: str
+    ) -> OliveModelHandler:
+        try:
+            from modelopt.onnx.quantization.int4 import quantize_int4  # type: ignore[import]
+        except ImportError as exc:
+            raise ImportError(
+                "Please install `olive-ai[nvmo]` or `nvidia-modelopt` to use INT4 AWQ quantization!"
+            ) from exc
+
+        data_dir = get_local_path_from_root(data_root, config["data_dir"])
+        calib_dataloader = self._user_module_loader.call_object(
+            config["dataloader_func"],
+            data_dir,
+            config["batch_size"],
+            config["calib_size"],
+            model_path=model.model_path,
+            **(config["dataloader_func_kwargs"] or {}),
+        )
+
+        if config["precision"] != "int4" or config["algorithm"] not in ["RTN", "AWQ"]:
```

Comment:
will other precisions be supported in the future? If not, then we could just remove the "precision" config parameter.

Can you add this check as part of `validate_search_point` like in https://github.com/microsoft/Olive/blob/3120697d4e9d2dee5af0877d9cfc082f6a422dfb/olive/passes/onnx/optimum_conversion.py#L53

Reply:
It is also possible to use a `str,Enum` as type for the param which comes with automatic validation like in https://github.com/microsoft/Olive/blob/3120697d4e9d2dee5af0877d9cfc082f6a422dfb/olive/passes/onnx/model_builder.py#L71

Reply:
Yes, other precisions will be supported as well. Looking into `validate_search_point `.